ksql-serde/src/main/java/io/confluent/ksql/serde/connect/ConnectSchemaTranslator.java

```diff
@@ -20,21 +20,43 @@
 import org.apache.kafka.connect.data.Field;
 import org.apache.kafka.connect.data.Schema;
 import org.apache.kafka.connect.data.SchemaBuilder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 public class ConnectSchemaTranslator {
+  private static final Logger log = LoggerFactory.getLogger(ConnectSchemaTranslator.class);
+
+  protected static class UnsupportedTypeException extends RuntimeException {
+    public UnsupportedTypeException(final String error) {
+      super(error);
+    }
+  }
+
   public Schema toKsqlSchema(final Schema schema) {
-    final Schema rowSchema = toKsqlFieldSchema(schema);
-    if (rowSchema.type() != Schema.Type.STRUCT) {
-      throw new KsqlException("KSQL stream/table schema must be structured");
+    final KsqlException nonStructRootError = new KsqlException(
+        "KSQL stream/table schema must be structured");
+    try {
+      final Schema rowSchema = toKsqlFieldSchema(schema);
+      if (rowSchema.type() != Schema.Type.STRUCT) {
+        throw nonStructRootError;
+      }
+      return rowSchema;
+    } catch (UnsupportedTypeException e) {
+      log.error("Unsupported type at root of schema: %s", e.getMessage());
```

Comment:
log and throw? Maybe we should just throw a `KsqlException` with the appropriate message and cause?